[OKTA-1153944] /revoke for AI agent token (#6131)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/ai-agent-token-exchange/main/resourceserver/revoke.md

After obtaining an OAuth STS access token through the token exchange flow, the AI agent can revoke that token using the org authorization server's `/oauth2/v1/revoke` endpoint and the same client credentials used for token exchange.

``` http
POST /oauth2/v1/revoke HTTP/1.1
Host: example.okta.com
Content-Type: application/x-www-form-urlencoded

token=eyJhbGciOiJIUzI1NiIsI...
&token_type_hint=oauth_sts
&client_assertion_type=urn:ietf:params:oauth:client-assertion-type:jwt-bearer
&client_assertion=eyJhbGci...
```

| Parameter | Description and value |
| --- | --- |
| `token` | The `access_token` value from the successful token exchange response. |
| `token_type_hint` | The value must be `oauth_sts`. |
| `client_assertion_type` | The value must be `urn:ietf:params:oauth:client-assertion-type:jwt-bearer`. |

> | `client_assertion` | The same signed JWT used for client authentication in the `/token` request. |

Reviewer: Sorry I think I was also not super clear in the JIRA description on this – the `client_assertion` is generated in the same way as it was for the `/token` request, but should be a newly generated JWT with the `/revoke` endpoint as the `aud` claim, and not the exact same JWT that was used.

Author: Not sure how I feel about how I phrased this, but let me know what you think @jamieguo-okta ! Alternatively, we could just use the same phrasing from before:

```
| client_assertion_type | The value must be `urn:ietf:params:oauth:client-assertion-type:jwt-bearer`. |
| client_assertion | A signed JWT used for client authentication. You must sign the JWT using the key created during the AI Agent registration. For more information on building the JWT, see [JWT with private key](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/client-auth/#jwt-with-private-key). |
```

Reviewer: hmm i think slight preference for reusing the same phrasing as the /token endpoint description but i'm fine w this too!

> | client_assertion |  A signed JWT used for client authentication. You must sign the JWT using the key created during the AI Agent registration. For more information on building the JWT, see [JWT with private key](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/client-auth/#jwt-with-private-key). |

Reviewer 2: ```suggestion
| client_assertion |  A signed JWT used for client authentication. You must sign the JWT using the key created during AI Agent registration. For more information on building the JWT, see [JWT with private key](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/client-auth/#jwt-with-private-key). |
```

> The endpoint returns `200 OK` regardless of whether the token was valid or already expired, to prevent information leakage.

Reviewer 2: ```suggestion
The endpoint returns `200 OK` regardless of whether the token was valid or already expired, to prevent information leaks.
```

> **Important:** Revoking an OAuth STS token at the Okta `/revoke` endpoint only removes the access token from Okta's token store. Keep the following behavior in mind:
>

> > - If Okta still holds a valid refresh token for the connection, the next token exchange request will obtain a new access token automatically.

Reviewer 2: ```suggestion
> - If Okta still holds a valid refresh token for the connection, the next token exchange request obtains a new access token automatically.
```
